spatialpandas/geometry/polygon2d.py

```diff
@@ -0,0 +1,163 @@
+        Returns:
+            spatialpandas Polygon2d
+        """
+        shape_parts = cls._shapely_to_coordinates(shape, orient)
+        return cls(shape_parts)
+
+    @property
+    def boundary(self):
+        # The representation of Polygon2dArray and MultiLine2dArray is identical
+        return MultiLine2d(self.data)
+
+    @property
+    def length(self):
+        return compute_line_length(self._values, self._value_offsets)
+
+    @property
+    def area(self):
+        return compute_area(self._values, self._value_offsets)
+
+
+class Polygon2dArray(GeometryArray):
+    _element_type = Polygon2d
+    _nesting_levels = 2
+
+    @property
+    def _dtype_class(self):
+        return Polygon2dDtype
+
+    @classmethod
+    def from_geopandas(cls, ga, orient=True):
+        """
+        Build a spatialpandas Polygon2dArray from a geopandas GeometryArray or
+        GeoSeries.
+
+        Args:
+            ga: A geopandas GeometryArray or GeoSeries of Polygon shapes.
+            orient: If True (default), reorder polygon vertices so that outer shells
+                    are stored in counter clockwise order and holes are stored in
+                    clockwise order.  If False, accept vertices as given. Note that
+                    some algorithms will not behave properly if the above ordering
+                    convention is not followed, so only set orient=False if it is
+                    known that this conventions is followed in the input data.
+        Returns:
+            Polygon2dArray
+        """
+        return cls([Polygon2d._shapely_to_coordinates(shape, orient) for shape in ga])
+
+    @property
+    def boundary(self):
+        # The representation of Polygon2dArray and MultiLine2dArray is identical
+        return MultiLine2dArray(self.data)
+
+    @property
+    def length(self):
+        result = np.full(len(self), np.nan, dtype=np.float64)
+        for c, result_offset in enumerate(self.offsets):
+            geometry_map_nested2(
+                compute_line_length,
+                result,
+                result_offset,
+                self.buffer_values,
+                self.buffer_offsets,
+                self.isna(),
+            )
+        return result
+
+    @property
+    def area(self):
+        result = np.full(len(self), np.nan, dtype=np.float64)
+        for c, result_offset in enumerate(self.offsets):
+            geometry_map_nested2(
+                compute_area,
+                result,
+                result_offset,
+                self.buffer_values,
+                self.buffer_offsets,
+                self.isna(),
+            )
+        return result
+
+
+def polygon_array_non_empty(dtype):
+    return Polygon2dArray(
+        [
+            [[1.0, 1.0, 2.0, 1.0, 2.0, 2.0, 1.0, 2.0, 1.0, 1.0],
+             [1.1, 1.1, 1.5, 1.9, 1.9, 1.1, 1.1, 1.1]],
+            [[1.0, 1.0, 2.0, 1.0, 2.0, 2.0, 1.0, 2.0, 1.0, 1.0]]
+        ], dtype=dtype
+    )
```

Comment:
This looks oddly specific for something not in tests/; maybe add a comment explaining what it's for?